README.md

```diff
@@ -1,2 +1,222 @@
+
+## Table of Contents
+- [Data Transformation](#data-transformation)
+    - [Workflow](#workflow)
+    - [dbt Project Structure](#dbt-project-structure)
+    - [Tests](#tests)
+- [Getting Started](#getting-started)
+    - [Oauth connection](#oauth)
+- [dbt Setup](#dbt-setup)    
+- [Working with dbt](#working-with-dbt)
+    - [Running Models](#running-models)
+    - [Running Tests](#running-tests)
+- [Project Structure](#project-structure)
+    - [Development Folders](#development-folders)
+
+
+## Data Transformation 
+
+dbt is composed of different moving parts working harmoniously. All of them are important to what dbt does — transforming data.  When you execute dbt run, you are running a model that will transform your data without that data ever leaving your warehouse.
+
+### Workflow
+
+The top level of a dbt workflow is the project. A project is a directory of a `.yml` file (the project configuration) and either `.sql` or `.py` files (the models). The project file tells dbt the project context, and the models let dbt know how to build a specific data set.
+
+A model is a single file containing a final select statement, and a project can have multiple models, and models can even reference each other.
+
+In dbt, you can configure the materialization of your models. Materializations are strategies for persisting dbt models in a warehouse. There are four types of materializations built into dbt. They are:
+1. Table (your model is rebuilt as a table on each run)
+2. View (your model is rebuilt as a view on each run)
+3. Incremental (allow dbt to insert or update records into a table since the last time that dbt was run.)
+4. Ephemeral (models are not directly built into the database. instead, dbt will interpolate the code from this model into dependent models as a common table expression.)
+
+### dbt Project Structure
+
+1. Staging (data preparation)
+The purpose of the source layer is to receive the raw data from the source and define preparations for further analysis.
+
+  What to do on staging:
+
+  - Column selection
+  - Consistently renaming columns.
+  - Definition of data types (casting columns to String, Numeric...)
+  - Flattening of structured objects
+  - Initial filters
+  - Tests (source.yml)
+  - Basic cleanup, like replacing empty strings with NULL, for example
+
+  What is not done in the source:
+
+  - Joins
+  - Creating business rules
+
+2. Intermediate (data preparation)
+In the intermediate layer, the preparation for directing the data to the marts takes place. Not every table in the staging layer will become an intermediate. In staging, it is possible to combine different tables to start assembling business rules.
+
+  What is done in the intermediate:
+
+  - Join between different source queries
+  - More complex functions that would not enter the staging layer
+  - Aggregations
+  - Creation of business metrics/rules
+
+  What not to do in intermediate:
+
+  - Opening of sources
```

Comment:
Opening of staging sounds clunky. Can we reword to something like "ingesting raw source data"